### PR TITLE
Add parsec provisionsing example

### PR DESCRIPTION
**Pull request by mohamed-arm**

### CODE REVIEW

File: host/examples/parsec_execute/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[package]
+authors = ["The Veracruz Development Team"]
+description = "The test program for provisioning a responder and executing a remote workload."
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/Cargo.toml]: Can you update the description to reflect what the program does?

File: host/examples/parsec_execute/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
+//! An example to send a file.
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: Same as above

File: host/examples/parsec_execute/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    let dpu_session_id = Session::from_url(dpu_server_url)?;
+
+    info!("Attesting DPU...");
+    attestation::request_attestation(dpu_session_id, proxy_attestation_server_url)?;
+    info!("Successfully attested DPU.");
+
+    info!("Preparing provisions...");
+    let filename = "my_parsec".to_owned();
+//    let data = b"bar".to_vec();
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: You can remove this now

File: host/examples/parsec_execute/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+use anyhow::anyhow;
+use log::{error, info};
+use utils::attestation;
+use transport::{messages::{Request, Response, Status}, session::Session};
+
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+
+fn get_file_as_byte_vec(filename: &String) -> Vec<u8> {
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: It is better practice to return an error than panicking. I recommend the use of anyhow errors (cf. main())

File: host/examples/parsec_execute/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    let dpu_server_url = DPU_SERVER_URL;
+    let proxy_attestation_server_url = ATTESTATION_SERVER_URL;
+
+    let dpu_session_id = Session::from_url(dpu_server_url)?;
+
+    info!("Attesting DPU...");
+    attestation::request_attestation(dpu_session_id, proxy_attestation_server_url)?;
+    info!("Successfully attested DPU.");
+
+    info!("Preparing provisions...");
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: What are you provisioning?

File: host/examples/parsec_execute/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    let filename = "my_parsec".to_owned();
+//    let data = b"bar".to_vec();
+    let data = get_file_as_byte_vec(&filename);
+
+    info!("Provisioning DPU...");
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: What are you provisioning?

File: host/examples/parsec_execute/src/main.rs [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        Response::Status(Status::Success(_)) => {
+            info!("Successfully provisioned DPU.");
+        },
+        _ => {
+            error!("Error provisioning DPU.");
+            return Err(anyhow!("Provisioning failure."));
+        },
+    }
+
+    info!("Requesting remote execution...");
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: Can you make the info! more descriptive by mentioning what you are executing?

File: host/examples/parsec_execute/src/main.rs [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        Response::Status(Status::Success(output)) => {
+            info!("Successfully executed code remotely. Output: \n{}", output);
+        },
+        _ => {
+            error!("Error executing code remotely.");
+            return Err(anyhow!("Remote execution failure."));
+        },
+    }
+
+    info!("Requesting remote execution #2");
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: Same here

**Review by gbryant-arm — CHANGES_REQUESTED** [review]
Looks good, just requires minor code and doc refactoring

File: host/examples/parsec_execute/src/main.rs [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+use transport::{messages::{Request, Response, Status}, session::Session};
+
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+
+const DPU_SERVER_URL: &str = "127.0.0.1:6666";
+const ATTESTATION_SERVER_URL: &str = "127.0.0.1:3010";
+
+const PARSEC_APP_NAME: &str = "parsec_app";
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: This its the parsec client's binary, right? Your model makes it difficult to understand and reproduce what code is actually provisioned and executed on the remote target.
How about you push the parsec-tool's binary instead and execute it over the CLI? Should solve most of your problems

File: host/examples/parsec_execute/src/main.rs [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        Response::Status(Status::Success(output)) => {
+            info!("Successfully executed code remotely. Output: \n{}", output);
+        },
+        _ => {
+            error!("Error executing code remotely.");
+            return Err(anyhow!("Remote execution failure."));
+        },
+    }
+
+    info!("Requesting p");
```
Comment (gbryant-arm) [inline host/examples/parsec_execute/src/main.rs]: "Requesting parsec execution"?

**Comment by mohamed-arm** [comment]
Addressed the comments.

**Review by gbryant-arm — APPROVED** [review]
LGTM

**Pull request merged**